harden-amz-linux.sh

```diff
@@ -1,8 +1,65 @@
 #!/bin/bash
+(
+set -xe #Exit on non-zero return and print commands & argruments
+
+# Set variables
+function usage()
+{
+  echo "ERROR: Incorrect arguments provided."
+  echo "Usage: $0 {args}"
+  echo "Where valid args are: "
+  echo "  -- nossh | Ignore ssh section"
+  echo "  -- nopassword | Ignore password section"
+  echo "  -- noaudit | Ignore audit section"
+  echo "  -- nonfs | Ignore NFS section"
+  echo "  -- noxwindows | Ignores X Windows" 
+  exit 1
+}
+
+POSITIONAL=()
+while [[ $# -gt 0 ]]
+do
+key="$1"
```

Comment:
When within a loop it's best to add white space to the beginning of the line to denote.